### PR TITLE
Fix session join requirements documentation

### PR DESCRIPTION
**Pull request by xacrimon**
Followup, please see context in #14149.

### CODE REVIEW

File: docs/pages/server-access/guides/tsh.mdx [hunk 1]
```diff
@@ -627,10 +627,23 @@ other user can join you through their terminal by typing:
...
 ```
 
-The Teleport user joining the session must have access to the same SSH principal
-used to start the session. In this case, Teleport user `joe` is logged in to the
-SSH Node as Unix user `joe`, so only Teleport users with a role allowing login `joe`
-will be able to join.
+The Teleport user joining the session must have a `join_sessions` policy set up that allows
+them to join sessions created by the other user. If `joe` has a role named `access`, the other
+user needs have a role with a policy as shown below.
+More detailed information about `join_sessions` policies can be found in the [role reference](../../access-controls/reference.mdx) and [moderated sessions guide](../../access-controls/guides/moderated-sessions.mdx).
```
Comment (ptgott) [inline docs/pages/server-access/guides/tsh.mdx:633]:
```suggestion
You can find more detailed information about `join_sessions` policies in the [role reference](../../access-controls/reference.mdx) and [Moderated Sessions guide](../../access-controls/guides/moderated-sessions.mdx).
```

File: docs/pages/server-access/guides/tsh.mdx [hunk 2]
```diff
@@ -627,10 +627,23 @@ other user can join you through their terminal by typing:
 $ tsh join <session_ID>
 ```
 
-The Teleport user joining the session must have access to the same SSH principal
-used to start the session. In this case, Teleport user `joe` is logged in to the
-SSH Node as Unix user `joe`, so only Teleport users with a role allowing login `joe`
-will be able to join.
+The Teleport user joining the session must have a `join_sessions` policy set up that allows
+them to join sessions created by the other user. If `joe` has a role named `access`, the other
+user needs have a role with a policy as shown below.
```
Comment (ptgott) [inline docs/pages/server-access/guides/tsh.mdx:632]:
```suggestion
user needs have a role with a policy as shown below:
```

File: docs/pages/server-access/guides/tsh.mdx (hunk 1 above)
Comment (ptgott) [inline docs/pages/server-access/guides/tsh.mdx:633]: I also think we should move this sentence into a separate paragraph below the code snippet, so we go directly from "as shown below" to the snippet.

**Review by ptgott — COMMENTED** [review]
Added some comments. At the same time, I'm not really sure if it's appropriate for this guide to include admin instructions related to setting up Teleport roles. If this guide is only for end users, the user running `tsh` will likely not have access to change their own roles. Is there another way for an end user to determine whether their cluster is set up to allow session joining without `tctl`/admin access?

**Review by zmb3 — APPROVED** [review]
thanks for this!

**Comment by xacrimon** [comment]
@ptgott I've addressed your comments. That said I don't really know of a better place for this. I think this guide is followed by a lot of people wanting to try the feature out, cluster admins and others.

**Comment by ptgott** [comment]
@xacrimon That's right—unless there's any other recourse for an end user without privileges to edit/list roles, I suppose the advice for end users would be "Tell your admin about this guide"

**Comment by xacrimon** [comment]
@zmb3 @ptgott In accordance with that plan, I've revised this and added an admonition for users lacking access referring them and their cluster administrator to the moderated sessions guide. I also added a quicklink in the introductory paragraph of the moderated sessions guide to make it easier to find the section on join policies.

File: docs/pages/access-controls/guides/moderated-sessions.mdx [hunk 3]
```diff
@@ -12,6 +12,9 @@ other users to be present in a Server or Kubernetes Access session. Depending on
 the requirements, these users can observe the session in real time, participate
 in the session, and terminate the session at will.
 
+In addition, Teleport administrators can define rules that allow users to join eachothers
```
Comment (ptgott) [inline docs/pages/access-controls/guides/moderated-sessions.mdx:15]:
```suggestion
In addition, Teleport administrators can [define rules](#join_sessions) that allow users to join each other's
```

File: docs/pages/access-controls/guides/moderated-sessions.mdx [hunk 4]
```diff
@@ -12,6 +12,9 @@ other users to be present in a Server or Kubernetes Access session. Depending on
...
+sessions from `tsh` and the Web UI. More on this [here](#join_sessions).
```
Comment (ptgott) [inline docs/pages/access-controls/guides/moderated-sessions.mdx:16]:
```suggestion
sessions from `tsh` and the Web UI.
```

**Review by ptgott — APPROVED** [review]

**Pull request merged**